### PR TITLE
DAOs migrated to use coroutines. RoomDataSources migrated to coroutines and return flow or entity.

### PR DESCRIPTION
**Pull request by mshaposhnyk**

### CODE REVIEW

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt [hunk 1]
```diff
@@ -11,58 +11,70 @@ import com.example.internshipgithubclient.db.user.UserRoomEntity
...
                 }
-                Observable.just(issue)
             }
             .map{
                 it.toDomain()
             }
             .toList()
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+            .observeOn(Schedulers.io())*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: Please remove commented code

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt [hunk 2]
```diff
@@ -7,60 +7,67 @@ import com.example.core.domain.User
...
+                    Observable.just(pull)
+                }
             }
             .map{
                 it.toDomain()
             }
             .toList()
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+            .observeOn(Schedulers.io())*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: Please remove commented code

File: app/src/main/java/com/example/internshipgithubclient/db/repo/RoomRepoDataSource.kt [hunk 3]
```diff
@@ -1,64 +1,72 @@
...
+        /*return repoDao.addRepoWatcher(repoWatcher)
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+            .observeOn(Schedulers.io())*/
+    }*/
+
+    override suspend fun deleteRepo(repo: Repo) {
+        return repoDao.deleteRepo(repo.fromDomain())
+        /*.subscribeOn(Schedulers.io())
+        .observeOn(Schedulers.io())*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/repo/RoomRepoDataSource.kt]: Please remove commented code

File: app/src/main/java/com/example/internshipgithubclient/db/user/RoomUserDataSource.kt [hunk 4]
```diff
@@ -44,20 +52,22 @@ class RoomUserDataSource(private val userDao: UserDao) : LocalUserDataSource {
...
-    override fun getAuthorized(): Single<User> {
-        return userDao.getAuthorizedUser()
+    override suspend fun getAuthorized(): User {
+        return userDao.getAuthorizedUser().toDomain()
+        /*return userDao.getAuthorizedUser()
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .map { it.toDomain() }
+            .observeOn(Schedulers.io())
+            .map { it.toDomain() }*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/user/RoomUserDataSource.kt]: Please remove commented code

**Review by mnilga — APPROVED** [review]

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt (hunk 1 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: fixed

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt (hunk 2 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: fixed

File: app/src/main/java/com/example/internshipgithubclient/db/repo/RoomRepoDataSource.kt (hunk 3 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/repo/RoomRepoDataSource.kt]: fixed

File: app/src/main/java/com/example/internshipgithubclient/db/user/RoomUserDataSource.kt (hunk 4 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/user/RoomUserDataSource.kt]: fixed

**Pull request merged**